.eslintrc.yml

```diff
@@ -0,0 +1,131 @@
+root: true
+
+parserOptions:
+  ecmaVersion: 2021
+  sourceType: module
+  project:
+    - ./tsconfig.json
+    - ./shared/tsconfig.json
+    - ./backend/tsconfig.json
+    - ./frontend/tsconfig.json
+
+settings:
+  typescript: true
+  node: true
+
+extends:
+  - eslint:recommended
+  - plugin:import/recommended
+  - plugin:import/typescript
+  - plugin:@typescript-eslint/recommended
+  - plugin:@typescript-eslint/recommended-type-checked
+  - plugin:@typescript-eslint/strict
+  - plugin:@typescript-eslint/strict-type-checked
+  - plugin:@typescript-eslint/stylistic
+  - plugin:@typescript-eslint/stylistic-type-checked
+  - plugin:unicorn/recommended
+  - plugin:sonarjs/recommended
+
+plugins:
+  - simple-import-sort
+
+rules:
+  no-restricted-syntax:
+    - error
+    - selector: SwitchCase > *.consequent[type!="BlockStatement"]
+      message: Switch cases without blocks are forbidden.
+    - selector: ExportAllDeclaration,ImportAllDeclaration
+      message: Export/Import all (*) is forbidden.
+    - selector: ExportNamedDeclaration[declaration!=null]
+      message: Exports should be at the end of the file.
+    - selector: TSEnumDeclaration,ClassDeclaration[abstract=true]
+      message: TS features are forbidden.
+    - selector: ImportDeclaration[importKind=type],ExportNamedDeclaration[exportKind=type]
+      message: Avoid import/export type { Type } from './module'. Prefer import/export { type Type } from './module'.
+  no-console:
+    - error
+  max-params:
+    - error
+    - 3
+  no-multiple-empty-lines:
+    - error
+    - max: 1
+  curly:
+    - error
+    - all
+  import/no-unresolved:
+    - off
+  import/extensions:
+    - error
+    - always
+    - ignorePackages: true
+  import/newline-after-import:
+    - error
+    - count: 1
+  import/no-default-export:
+    - error
+  '@typescript-eslint/padding-line-between-statements':
+    - error
+    - blankLine: never
+      prev: export
+      next: export
+    - blankLine: always
+      prev:
+        - const
+        - class
+      next: export
+  '@typescript-eslint/consistent-type-definitions':
+    - off
+  '@typescript-eslint/non-nullable-type-assertion-style':
+    - off
+  '@typescript-eslint/return-await':
```

Comment:
not sure if I need to add any additional rules at the beginning of the project. they are usually added as needed

Reply:
This is needed from the start, it requires to use ! instead of casting, or possible null types
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/non-nullable-type-assertion-style.md